src/back/hlsl/writer.rs

```diff
@@ -1331,6 +1353,62 @@ impl<'a, W: Write> Writer<'a, W> {
 
         Ok(())
     }
+
+    /// Helper function that write default zero initialization for supported unitialized types
+    fn write_default_init(&mut self, module: &Module, ty: Handle<crate::Type>) -> BackendResult {
```

Comment:
sorry, I still don't understand why we have this function.
Zero-initialization should be as simple as:
```cpp
write!(self.out, "(")?;
self.write_type(ty)?;
write!(self.out, ")0")?;
```

Reply:
Sorry, I missed the fact, that it can that simple. Thank you!